ncm-gpfs/src/main/perl/gpfs.pm

```diff
@@ -265,6 +263,26 @@ sub runcurl {
     return $output || 1;
 };
 
+sub reinstall_update_rpms {
+    my ($self, $config, $rpms) = @_;
+    my $ret = 1;
+    my $useyum =  $config->getValue("$mypath/base/useyum");
+    my @reinrpms = @$rpms;
+
+    if (scalar @reinrpms) {
+        if ($useyum) {
+            # for dependencies
+            $self->runyum($config, "install", @reinrpms) || return;
```

Comment:
don't return here. for both `runyum` and `runcmd`, do a `$ret = $self->run...`; and end this block with
```perl
if($ret) {
    $self->info("Rpms reinstalled");
} else {
   $self->error("Reinstalling rpms failed");
}
```